datamodel/include/datamodel/math/Vector.hpp

```diff
@@ -218,7 +222,9 @@ class Vector
 using Vector4d = Vector<core::EngineScalar, 4>;
 using Vector3d = Vector<core::EngineScalar, 3>;
 using Vector2d = Vector<core::EngineScalar, 2>;
+using Vector4f = Vector<float, 4>;
```

Comment:
If `core::EngineScalar` is redefined to be `float`, we will have basically the same types. I think we should think of a way how to handle this cases.

Reply:
Can't see how this is a problem. It will just mean we can pass d variants to float variants, since they will be the same type. If you mean that there should be an alias for `float` I don't think that is right either, since we shouldn't be making aliases for all the built-in types.